lib/Raven/Client.php

```diff
@@ -747,32 +735,10 @@ public function send(&$data)
             return;
         }
 
-        $message = $this->encode($data);
-
-        $headers = [
-            'User-Agent' => static::getUserAgent(),
-            'X-Sentry-Auth' => $this->getAuthHeader(),
-            'Content-Type' => 'application/octet-stream'
-        ];
-
-        $config = $this->getConfig();
-        $server = sprintf('%s/api/%d/store/', $config->getServer(), $config->getProjectId());
-
-        $this->send_remote($server, $message, $headers);
-    }
+        $uri = sprintf('api/%d/store/', $this->getConfig()->getProjectId());
+        $request = $this->requestFactory->createRequest('POST', $uri, [], JSON::encode($data));
 
-    /**
-     * Send data to Sentry
-     *
-     * @param string       $url     Full URL to Sentry
-     * @param array|string $data    Associative array of data to log
-     * @param array        $headers Associative array of headers
-     */
-    protected function send_remote($url, $data, $headers = [])
-    {
-        $parts = parse_url($url);
-        $parts['netloc'] = $parts['host'].(isset($parts['port']) ? ':'.$parts['port'] : null);
-        $this->send_http($url, $data, $headers);
+        $this->httpClient->sendAsyncRequest($request)->wait(true);
```

Comment:
>So an option to choose between sync and async would be ideal?

There was an option that configured this in 1.x named `curl_method`, I removed it because we're now relying on the HTTP client itself implementing `HttpClient`, `HttpAsyncClient` or both interfaces to decide which method should be used to send events. What I didn't think of is the last case: which method is preferred in case both interfaces are implemented by the client?